### PR TITLE
Filter out instances with state = TERMINATED

### PR DESCRIPTION
**Pull request by ocsw**
**What this PR does / why we need it**:
Since we're seeing periodic janitor failures with the TERMINATED filter in config, let's try filtering them out in code.

**Which issue(s) this PR fixes**

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Filter out instances with state = TERMINATED in the authority.
```

### CODE REVIEW

**Comment by ocsw** [comment]
@spotify/alf PTAL

**Comment by ocsw** [comment]
I got that doc problem locally too but I thought it might be a network issue.  Looks like something we're relying on vanished?  Although I was able to download _something_ when I went to the URL, then it gave me a 404 page?

**Comment by ocsw** [comment]
(@descent3me fixed that in another PR)

File: src/gordon_gcp/plugins/janitor/authority.py [hunk 1]
```diff
@@ -174,8 +174,15 @@ def _filter_results(self, results):
...
-
-        return self._filter_results(all_results)
+        all_results = self._filter_results(all_results)
+
+        instances = []
+        for instance in all_results:
+            if instance['status'] == 'TERMINATED':
+                logging.info(f'Skipping terminated instance: "{instance}"')
+                continue
+            instances.append(instance)
```
Comment (econchick) [inline src/gordon_gcp/plugins/janitor/authority.py]: Any harm in moving this up to the `_filter_results` method? We seem to be doing some skipping up there.

Comment (ocsw) [reply src/gordon_gcp/plugins/janitor/authority.py]: I was thinking this is better grouping of functionality - filter results filters the results themselves, get instances gets (filtered) instances?

File: tests/unit/plugins/janitor/test_authority.py [hunk 2]
```diff
@@ -232,3 +236,11 @@ def test_create_msgs_bad_json(caplog, fake_authority):
     results = fake_authority._create_msgs([partial_instance])
     assert [] == results
     assert 1 == len(caplog.records)
+
+
+# test_get_instances_terminated(mocker, authority_config, metrics,
```
[the remark this answers is not in the document]
Comment (ocsw) [reply tests/unit/plugins/janitor/test_authority.py]: oops

**Pull request merged**